ca_candidates/people.py

```diff
@@ -0,0 +1,247 @@
+                .replace("ô", "o")
+                .replace("ü", "u")
+                .lower()
+            )
+            url = "https://" + nameclean + ".ndp.ca"
+            if name in {
+                "Julie Girard-Lemay",
+                "Tommy Bureau",
+                "Lise  Garon",
+                "Hugues Boily-Maltais",
+            }:  # these candidates have npd instead of ndp in url
+                url = "https://" + nameclean + ".npd.ca"
+            if name == "Arlington Antonio Santiago":
+                url = "https://arlingtonsantiago.ndp.ca"
+            if name == "Tammy Bentz":
+                url = url + "/contact"
+
+            p.add_source(NDP_PAGE)
+            p.add_source(url)
+
+            candidatepage = self.lxmlize(url)
+
+            email_el = candidatepage.xpath('//a[contains(@href, "mailto:")]/@href')
+            if email_el:
+                email = email_el[0].replace("mailto:", "")
+                p.add_contact("email", email)
+
+            facebook_el = candidatepage.xpath('//a[contains(@href, "facebook") or contains(@href, "fb")]/@href')
+            if facebook_el:
+                facebook = facebook_el[0]
+                p.add_link(facebook)
+
+            instagram_el = candidatepage.xpath('//a[contains(@href, "instagram")]/@href')
+            if instagram_el:
+                instagram = instagram_el[0]
+                p.add_link(instagram)
+
+            twitter_el = candidatepage.xpath('//a[contains(@href, "twitter")]/@href')
+            if twitter_el:
+                twitter = twitter_el[0]
+                p.add_link(twitter)
+
+            youtube_el = candidatepage.xpath('//a[contains(@href, "youtube")]/@href')
+            if youtube_el:
+                youtube = youtube_el[0]
+                p.add_link(youtube)
+
+            phone_el = candidatepage.xpath('//a[contains(@href, "tel:")]/@href')
+            if phone_el:
+                phone = phone_el[0].replace("tel:", "")
+                p.add_contact("voice", phone, "office")
+
+            yield p
+
+    def scrape_liberal(self):
+        page = self.lxmlize(LIBERAL_PAGE)
+
+        candidates = page.xpath('//div[@class="person-listing-container"]/article')
+        assert len(candidates)
+
+        for candidate in candidates:
+            name = candidate.xpath("./div/header/h2/text()")[0]
+            district = (
+                candidate.xpath("./div/header/h3/text()")[0]
+                .replace("- ", "—")
+                .replace(" – ", "—")
+                .replace("–", "—")
+                .replace("\u200f", "")
+                .replace("\u2013", "-")
+                .replace("\u2014", "-")
+                .strip()
+            )
+            district = self.normalize_district(district)
+            if district == "Surrey-Newton":
+                district = "Surrey Newton"
+            d = self.normalized_names[district]
+            if d == "Mont-Saint-Bruno—L’Acadie":
+                d = "24049"
```

Comment:
@jpmckinney I couldn't figure out anything better than this but not 100% sure if a better way to resolve this problem